agent/util/archive_tar.go

```diff
@@ -23,117 +23,99 @@ func BuildArchive(ctx context.Context, tarWriter *tar.Writer, rootPath string, i
 	pathsToAdd := streamArchiveContents(ctx, rootPath, includes, []string{})
 
 	numFilesArchived := 0
-	done := make(chan bool)
-	errChan := make(chan error)
-	go func(inputChan <-chan ArchiveContentFile) {
-		defer func() {
-			errChan <- recovery.HandlePanicWithError(recover(), nil,
-				"building archive")
-		}()
-		processed := map[string]bool{}
-		logger.Infof("beginning to build archive")
-	FileChanLoop:
-		for file := range inputChan {
-			if ctx.Err() != nil {
-				return
-			}
+	defer func() {
+		grip.Alert(recovery.HandlePanicWithError(recover(), nil,
```

Comment:
Before if there was an error we added it to errChan and then returned it, but here it's only being logged. Won't this just suppress the problem?

Reply:
Yep, looks like previously we would have retried if this panicked. I'm going to remove this panic handler and let the one 2 stacks below this handle the panic, which also retries